apis/submariner/v1alpha1/submariner_types.go

```diff
@@ -52,6 +52,8 @@ type SubmarinerSpec struct {
 	CeIPSecIKEPort           int                  `json:"ceIPSecIKEPort,omitempty"`
 	CeIPSecNATTPort          int                  `json:"ceIPSecNATTPort,omitempty"`
 	CeIPSecDebug             bool                 `json:"ceIPSecDebug"`
+	CeIPSecPreferredServer   *bool                `json:"ceIPSecPreferredServer,omitempty"`
+	CeIPSecForceUDPEncaps    *bool                `json:"ceIPSecForceUDPEncaps,omitempty"`
```

Comment:
Why are these pointers? They can be plain `bool` AFAICT (false when absent).

Reply:
to allow the "omitEmpty", otherwise we can omit them, or can we?

Reply:
Otherwise they will get added to https://github.com/submariner-io/submariner-operator/pull/1200/files#diff-3bf1b6c02ff8069cea7b4c42751006ce5baafde160d68e6272782f726e9c738bL111

Reply:
good point.

Reply:
Do we need to omit if empty?

Reply:
`bool` values can be `omitempty` without being pointers, they are then omitted if false (see https://golang.org/pkg/encoding/json/#Marshal).

Reply:
I wanted to avoid issues on upgrade of the operator for example, but may be there is no such issue?

Reply:
This would also fix the alignment linting error ;-).

Reply:
For example the ACM team using the operator alone will need to start providing those new fields, but they are optional. I believe it's better to keep with omitEmpty / non-required

Reply:
There shouldn't be. On deserialization from previous version, if the field is missing it will be false, exactly if it was omitted if empty.

Reply:
No, they wouldn’t. Absent values would be unmarshalled as false. (I’m building to check, just to make sure.)

Reply:
So `make manifests` just finished with

```suggestion
	CeIPSecPreferredServer   bool                 `json:"ceIPSecPreferredServer,omitempty"`
	CeIPSecForceUDPEncaps    bool                 `json:"ceIPSecForceUDPEncaps,omitempty"`
```

and the CRDs stay the same: the values aren’t required, and old CRs can still be unmarshalled.

Reply:
> No, they wouldn’t. Absent values would be unmarshalled as false. (I’m building to check, just to make sure.)

This was in reply to Miguel, not Tom ;-).

Reply:
> > No, they wouldn’t. Absent values would be unmarshalled as false. (I’m building to check, just to make sure.)
> 
> This was in reply to Miguel, not Tom ;-).

I know 😄

Reply:
See `ServiceDiscoveryEnabled` for an existing example of this, just below :smile:

Reply:
Oh I was so sure that without omitEmpty they become "required", I'll update the patch in the next minutes, thanks!

Reply:
I mean, I was sure that without making it a pointer it would fail... 'XD

Reply:
Done.